File: MathAndPrimeNumbers/CountPrimes/countPrime.cpp
#include <vector>
#include <cmath>
using namespace std;

class Solution {
public:
    void fillSieve(vector<bool>& sieve){
        int n = sieve.size();
        for(int i=2;i<=sqrt(n);i++){
            if(sieve[i]){
                for(int j=i*i;j<n;j+=i){
                    sieve[j] = false;
                }
            }
        }
        return;
    }
    int countPrimes(int n) {
        if(n<=2)return 0;
        n = n-1;
        vector<bool>sieve(n+1,true);
        sieve[0] = false;
        sieve[1] = false;
        fillSieve(sieve);
        int totalPrimes = 0;
        for(int i=0;i<=n;i++){
            if(sieve[i])totalPrimes++;
        }
        return totalPrimes;
    }
};

File: MathAndPrimeNumbers/CountPrimes/countPrime.py
from typing import List
from math import sqrt

class Solution:
    def fillSieve(self, sieve: List):
        n = len(sieve)
        for i in range(2,int(sqrt(n)+1)):
            if sieve[i]:
                for j in range(i*i,n,i):
                    sieve[j] = False

    def countPrimes(self, n: int) -> int:
        if n<=2: return 0
        n = n-1
        sieve = [True] * (n+1)
        sieve[0] = False
        sieve[1] = False
        self.fillSieve(sieve)
        totalPrimes = 0
        for i in range(n+1):
            if sieve[i]: totalPrimes += 1
        return totalPrimes